### PR TITLE
feat(metrics): added collectMetricsExpressMiddleware and deprecated defaultMetricsMiddleware

### PR DESCRIPTION
**Pull request by syncush**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✔                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,39 @@
...
+### [5.1.4](https://github.com/MapColonies/telemetry/compare/v5.1.3...v5.1.4) (2023-07-31)
+
+### [5.1.3](https://github.com/MapColonies/telemetry/compare/v5.1.2...v5.1.3) (2023-07-31)
+
+### [5.1.2](https://github.com/MapColonies/telemetry/compare/v5.1.1...v5.1.2) (2023-07-31)
+
+### [5.1.1](https://github.com/MapColonies/telemetry/compare/v5.1.0...v5.1.1) (2023-07-31)
+
+## [5.1.0](https://github.com/MapColonies/telemetry/compare/v5.0.0...v5.1.0) (2023-07-31)
+
```
Comment (CptSchnitz) [inline CHANGELOG.md:30]: ?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@map-colonies/telemetry",
-  "version": "5.0.0",
+  "version": "5.1.13",
```
Comment (CptSchnitz) [inline package.json:3]: ?

File: src/metrics/middleware/metrics.ts [hunk 3]
```diff
@@ -1,5 +1,30 @@
...
+    return null;
+  }
+  return {
+    major: match[1],
+    minor: match[2],
+    patch: match[3],
+    prerelease: match[4],
+    build: match[5],
+  };
+};
```
Comment (CptSchnitz) [inline src/metrics/middleware/metrics.ts:19]: move to a common util area

File: src/metrics/middleware/metrics.ts [hunk 4]
```diff
@@ -19,3 +44,78 @@ export function metricsMiddleware(
...
+      res.end(await register.metrics());
+    } catch (error) {
+      return next(error);
+    }
+  };
+}
+
+export function collectMetricsExpressMiddleware(options: Partial<Opts>): promBundle.Middleware {
+  const pacakgeInfo = loadPackageInfo();
+  const defaultOpts: Opts = {
```
Comment (CptSchnitz) [inline src/metrics/middleware/metrics.ts:66]: maybe upgrade typescript version and try the satisfy keyword

File: src/metrics/middleware/metrics.ts [hunk 5]
```diff
@@ -19,3 +44,78 @@ export function metricsMiddleware(
...
+  };
+  const meregedOptions = { ...defaultOpts, ...options };
+  /* eslint-disable @typescript-eslint/naming-convention */
+  const mergedLabels = {
+    hostname: osHostname(),
+    service_name: pacakgeInfo.name,
+    ...(options.labels ? options.labels : {}),
+  };
+  register.setDefaultLabels(mergedLabels);
+  if (meregedOptions.collectNodeMetrics) {
```
Comment (CptSchnitz) [inline src/metrics/middleware/metrics.ts:81]: merged

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/metrics/middleware/metrics.ts (hunk 5 above)
Comment (syncush) [reply src/metrics/middleware/metrics.ts:81]: done

File: src/metrics/middleware/metrics.ts (hunk 3 above)
Comment (syncush) [reply src/metrics/middleware/metrics.ts:19]: done

File: package.json (hunk 2 above)
Comment (syncush) [reply package.json:3]: done

File: src/metrics/middleware/metrics.ts (hunk 4 above)
Comment (syncush) [reply src/metrics/middleware/metrics.ts:66]: done

**Pull request merged**